### ISSUE TITLE
Focus visible for small elements with large click areas

### ISSUE THREAD

**Issue by alastc** [w3c/wcag#1004]
An issue @jake-abma raised on email:
<hr>
Please see code below where we use web component structures where the parent / main container, click-area and button are separated and focus indicators are made to fit OR buttons OR placement (like top right corner in dialog OR on touch target area (depending on need of design)

Depending on the 'button' as "user-interface component" is also not fine as we do not use them visually (for visual purposes), can even make them 1px by 1px and so the focus indicator of 2 CSS px would be fine?

![image](https://user-images.githubusercontent.com/216630/71680483-24864880-2d82-11ea-8da7-74a5001bf609.png)

Interaction can be different things:

- for keyboard behaviour the button element is still used
- visual interaction (communication / with colors / icons) can be the <button> element OR the outer element OR click area
- with mouse the click area, is adjustable

**Comment by alastc** [comment]
Hi @jake-abma I put together some more examples for testing & working out how it works:
https://alastairc.uk/tests/wcag22-examples/focus-more-visible-4.html 

I think examples 21 & 22 demontrate what you were aiming to show, I put them right on the edge of the criteria text.

**Comment by jake-abma** [comment]
Thanks @alastc indeed a good one, and this knocks the SC down, as this is weird when the indicator is enough for the button, but with the extra touch target, fails the same button (visually)

**Comment by jake-abma** [comment]
We might see if we can blend the current calculation / SC text with a kind of 'baseline', preventing this discrepancy

**Comment by alastc** [comment]
Rather than a baseline approach, I think it would be better to make an exception where there is a visible boundary smaller than the element.

It is good that the standard way of doing focus is outlines/borders/backgrounds, and generally they would pass in the examples. It is only because the examples are at the minumum it's an issue.

Anyway, for those type of examples, how about adding:

> Note: If the control has a visible boundary smaller than the hit area, the size measure is taken from the visible boundary.

**Comment by mraccess77** [comment]
+1 to Alastair's suggested note to address the issue.

**Comment by detlevhfischer** [comment]
+1 yes, good exception to cover cases of larger click areas

**Comment by alastc** [comment]
Note added to the proposed update ([version 3 in this doc](https://docs.google.com/document/d/1F83m-HXRkXz1QCF6_QNtQGzIULugMiFSLgtqDRgKA-s/edit#heading=h.96ib39rkbmhb)), does that suffice @jake-abma

**Comment by jake-abma** [comment]
yes! thanks you
